cdap-api/src/main/java/co/cask/cdap/api/dataset/table/RowColumns.java

```diff
@@ -43,16 +48,19 @@
 
   public RowColumns(byte[] row) {
     this.row = row;
-    this.columns = new ArrayList<>();
   }
 
   public RowColumns(byte[] row, byte[]... columns) {
     this(row);
+    this.columns = new ArrayList<>();
     add(columns);
   }
 
   @SuppressWarnings("unchecked")
   public T add(byte[]... columns) {
+    if (this.columns == null) {
```

Comment:
This actually mean if someone do this:

```java
Delete delete = new Delete(row);  // Delete all columns
delete.add();    // It is valid to be called without any arg, but this turn the delete to delete no column
```

I think it's better to guard against empty arg call by having the signature as `T add(byte[] firstColumn, byte[]... moreColumns)`